UnityProject/Assets/Scripts/Objects/Research/Artifacts/Artifact.cs

```diff
@@ -9,182 +18,469 @@ public class ArtifactSprite
+
+		[SyncVar] public ArtifactData artifactData = new ArtifactData();
+
+		public ArtifactDataSO ArtifactDataSO;
+
+		public AreaEffectBase AreaEffect;
+		public InteractEffectBase InteractEffect;
+		public DamageEffectBase DamageEffect;
+
+
+		[SyncVar] public string ID = "T376";
+
+		public bool UnderTimeoutTouch
 		{
-			// check that timeout has passed
-			return Time.time - lastActivationTime < EffectTimeout;
+			get
+			{
+				// check that timeout has passed
+				return Time.time - lastActivationTimeTouch < TouchEffectTimeout;
+			}
 		}
-	}
 
-	private void Awake()
-	{
-		currentEffect = GetComponent<ArtifactEffect>();
-	}
+		public bool UnderTimeoutDamage
+		{
+			get
+			{
+				// check that timeout has passed
+				return Time.time - lastActivationTimeDamage < DamageEffectTimeout;
+			}
+		}
 
-	private void OnEnable()
-	{
-		UpdateManager.Add(CallbackType.UPDATE, UpdateMe);
-	}
+		#region Lifecycle
 
-	private void OnDisable()
-	{
-		UpdateManager.Remove(CallbackType.UPDATE, UpdateMe);
-	}
+		private void Awake()
+		{
+			integrity = GetComponent<Integrity>();
+			radiationProducer = GetComponent<RadiationProducer>();
+			objectPhysics = GetComponent<UniversalObjectPhysics>();
 
-	private void UpdateMe()
-	{
-		if (!CustomNetworkManager.IsServer)
+			integrity.OnApplyDamage.AddListener(DoDamageEffect);
+		}
+
+		public void OnSpawnServer(SpawnInfo info)
 		{
-			return;
+			//Sets appearance
+			artifactData.Type = (ArtifactType)Random.Range(0, 3);
+
+			//Select random sprite
+			ServerSelectRandomSprite();
+
+			//Add it to spawned artifacts registry for artifact detector
+			if (!ServerSpawnedArtifacts.Contains(this))
+				ServerSpawnedArtifacts.Add(this);
+
+			ArtifactClass Compostion;
+
+			//Add elements to the artifacts compisition 
+			for (int i = 0; i < Random.Range(1, 5); i++)
+			{
+				//Choose what the artifact should be made of
+				Compostion = (ArtifactClass)Random.Range(0, 3);
+
+				//Adds properties to the artifact depending on what material has been added
+				switch (Compostion)
+				{
+					case ArtifactClass.Uranium:
+						artifactData.radiationlevel += Random.Range(1000, 5000);
+						break;
+					case ArtifactClass.Bluespace:
+						artifactData.bluespacesig += Random.Range(50, 150);
+						break;
+					case ArtifactClass.Bananium:
+						artifactData.bananiumsig += Random.Range(100, 500);
+						break;
+					default:
+						artifactData.radiationlevel += Random.Range(1000, 5000);
+						break;
+				}
+			}
+
+			//Randomises ID and adds it to name, this is done for two reasons:
+			//To prevent duplication exploits, server knows if youve researched an artifact before and won't give you extra RP or credits for repeated research.
+			//So players know which artifact is which if they have the same sprites
+			//ID is in form: A000 - Z999
+			int num = Random.Range(0, 26);
+			ID = $"{(char)('a' + num)}{Random.Range(0, 1000).ToString("000")}";
+			ID = ID.ToUpperInvariant();
```

Comment:
If you're going to rely on this for other things, I would suggest at least checking for collisions.